src/transformers/generation/utils.py

```diff
@@ -1975,6 +1990,7 @@ def contrastive_search(
                     model_kwargs,
                     is_encoder_decoder=self.config.is_encoder_decoder,
                     standardize_cache_format=True,
+                    model_inputs=model_inputs,
```

Comment:
we need to update `model_kwargs` with `model_inputs["cache_position"]`, hence the additional argument